### PR TITLE
Add binding informations

### PR DESCRIPTION
**Pull request by jdneo**
cache another two metadata file into the maven plugin:
- bindings.json: stores the parameter information
- Resources.json: stores the string constants

All the `templates.json`, `bindings.json` & `Resources.json` will be printed into system out, and be consumed by VS Code extension.

### CODE REVIEW

File: azure-functions-maven-plugin/pom.xml [hunk 1]
```diff
@@ -246,6 +246,8 @@
                 <includes>
                     <include>**/*.xml</include>
                     <include>templates.json</include>
+                    <include>bindings.json</include>
+                    <include>Resources.json</include>
```
Comment (akaroml) [inline azure-functions-maven-plugin/pom.xml]: Is there any reason we capitalize the first letter?

File: azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/ListMojo.java [hunk 2]
```diff
@@ -18,17 +19,37 @@
...
+
+    protected static final String BINDINGS_START = ">> bindings begin <<";
+    protected static final String BINDINGS_END = ">> bindings end <<";
+
+    protected static final String RESOURCES_START = ">> resources begin <<";
+    protected static final String RESOURCES_END = ">> resources end <<";
+
+    protected static final String TEMPLATES_FILE = "/templates.json";
+    protected static final String BINDINGS_FILE = "/bindings.json";
+    protected static final String RESOURCES_FILE = "/Resources.json";
```
Comment (akaroml) [inline azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/ListMojo.java]: Same question.

File: azure-functions-maven-plugin/src/main/resources/Resources.json [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    "cosmosDBIn_databaseName_label": "Database name",
+    "cosmosDBIn_databaseName_help": "Name of the Cosmos DB database that includes the collection to be monitored.",
+    "cosmosDBIn_collectionName_label": "Collection name",
+    "cosmosDBIn_collectionName_help": "Name of the collection to be monitored.",
+    "cosmosDBIn_leaseCollectionName_label": "Collection name for leases",
+    "cosmosDBIn_leaseCollectionName_help": "Name of the collection to store the leases.",
+    "cosmosDBIn_createIfNotExists_label": "Create lease collection if it does not exist",
+    "cosmosDBIn_createIfNotExists_help": "Checks for existence and automatically creates the leases collection."
+  }
+}
```
Comment (akaroml) [inline azure-functions-maven-plugin/src/main/resources/Resources.json]: There needs to be a blank line at the end of the file.

File: azure-functions-maven-plugin/src/main/resources/bindings.json [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+          "value": "boolean",
+          "defaultValue": true,
+          "required": true,
+          "label": "$cosmosDBIn_createIfNotExists_label",
+          "help": "$cosmosDBIn_createIfNotExists_help"
+        }
+      ]
+    }
+  ]
+}
```
Comment (akaroml) [inline azure-functions-maven-plugin/src/main/resources/bindings.json]: Add a blank line here.

File: azure-functions-maven-plugin/src/main/resources/Resources.json [hunk 5]
```diff
@@ -0,0 +1,65 @@
+{
+  "$schema": "<TBD>",
+  "contentVersion": "1.0.0",
```
Comment (akaroml) [inline azure-functions-maven-plugin/src/main/resources/Resources.json]: "version" should be enough.

File: azure-functions-maven-plugin/src/main/resources/Resources.json [hunk 6]
```diff
@@ -0,0 +1,65 @@
+{
+  "$schema": "<TBD>",
```
Comment (akaroml) [inline azure-functions-maven-plugin/src/main/resources/Resources.json]: So the TBD will be filled with some logical name like "resources". Or will it reference some json schema that is stored in some schema store?

File: azure-functions-maven-plugin/pom.xml (hunk 1 above)
Comment (jdneo) [reply azure-functions-maven-plugin/pom.xml]: These file names are following the pattern of the existing templates files.

**Pull request merged**